src/ape/api/providers.py

```diff
@@ -403,6 +404,29 @@ def get_contract_logs(self, log_filter: LogFilter) -> Iterator[ContractLog]:
             Iterator[:class:`~ape.types.ContractLog`]
         """
 
+    def send_private_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
+        """
+        Send a transaction through a private mempool (if supported by the Provider).
+
+        Raises:
+            :class:`~ape.exceptions.APINotImplementedError`: If using a non-local
+              network and not implemented by the provider.
+
+        Args:
+            txn (:class:`~ape.api.transactions.TransactionAPI`): The transaction
+              to privately publish.
+
+        Returns:
+            :class:`~ape.api.transactions.ReceiptAPI`
+        """
+        if self.network.name == LOCAL_NETWORK_NAME or self.network.name.endswith("-fork"):
```

Comment:
Maybe add a warning "`private=True` is set, but bypassing private mempool because we are on a local network"